common/djangoapps/student/tests/test_email.py

```diff
@@ -623,6 +622,6 @@ def _assert_email(self, subject, body_fragments):
 
         assert message.subject == subject
 
-        for body in text, html:
-            for fragment in body_fragments:
-                assert fragment in body
+        for fragment in body_fragments:
+            assert fragment in text
+            assert escape(fragment) in html
```

Comment:
Why you are escaping here?

Reply:
Due to these changes https://github.com/edx/edx-platform/pull/24021/files#diff-021093b57b61de863c2688df79f959d0R13-R19, because these were not being escaped before and we were simply asserting unescaped text for both text and html templates but after escaping these strings the html content will have these escaped strings, so we have to escape our test data before asserting otherwise it will fail.